### PR TITLE
Use mount namespace for faster tracking

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
This allows faster PID tracking via the mount namespace. The namespace
gets passed via the event loop and can be used to identify already
tracked profiles.
#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@pjbgf @JAORMX @jhrozek PTAL. This makes tracking short-lived PIDs possible.

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go [hunk 1]
```diff
@@ -447,19 +480,50 @@ func (b *BpfRecorder) processEvents(events chan []byte) {
...
-			b.logger.Info("Saving PID for profile", "pid", pid, "profile", profile, "comm", comm)
-
-			pids, _ := b.pidsForProfiles.LoadOrStore(profile, []Pid{})
-			b.pidsForProfiles.Store(profile, append(pids.([]Pid), Pid{id: pid, comm: comm}))
 		}
 	}
 }
 
+func (b *BpfRecorder) findSystemMountNamespace() (uint64, error) {
+	res, err := b.Readlink("/proc/1/ns/mnt")
```
Comment (pjbgf) [inline internal/pkg/daemon/bpfrecorder/bpfrecorder.go:496]: May be worth adding a comment highlighting that this requires `hostPID`.

**Review by pjbgf — COMMENTED** [review]
nice one, just a comment for a follow-up PR.

/lgtm

File: internal/pkg/daemon/bpfrecorder/bpfrecorder.go (hunk 1 above)
Comment (saschagrunert) [reply internal/pkg/daemon/bpfrecorder/bpfrecorder.go:496]: Following-up in https://github.com/kubernetes-sigs/security-profiles-operator/pull/647

**Pull request merged**